CMakeLists.txt

```diff
@@ -15,6 +15,17 @@ option(BUILD_TESTING "Run unit tests" OFF)
 option(CMAKE_RUN_CLANG_TIDY "Run clang-tidy" OFF)
 option(LIBOCPP_BUILD_EXAMPLES "Build charge_point and central_system binaries." OFF)
 option(OCPP_INSTALL "Install the library (shared data might be installed anyway)" ${EVC_MAIN_PROJECT})
+option(USE_NON_STD_FILESYSTEM "Usage of boost/filesystem.hpp instead of std::filesystem" OFF)
```

Comment:
I would suggest we name this option 
```cmake
option(LIBOCPP_USE_BOOST_FILESYSTEM "Usage of boost/filesystem.hpp instead of std::filesystem" OFF)
```
To make it prefixed to just this library and to clarify the function of it. I would rather rename this option later on if we choose to change the underlying implementation again

Reply:
Did you see my [last comment on this in the comment above](https://github.com/EVerest/libocpp/pull/177#discussion_r1331797400)?

Reply:
Yes but I would disagree in this instance and strongly recommend the usage of the prefixed variable LIBOCPP_USE_BOOST_FILESYSTEM, we can always set a few of these variables in everest-core if needed but this makes it easier to specifically enable/disable this feature lib-by-lib

Reply:
Ok, makes sense. :+1: